custom_components/divoom_pixoo/sensor.py

```diff
@@ -73,112 +70,121 @@ async def async_will_remove_from_hass(self):
             return
 
-        def update():
-            pixoo = self._pixoo
-            pixoo.clear()
-
-            if "channel" in self.page:
-                for ch in self.page["channel"]:
-                    pixoo.set_custom_page(ch['number'])
-
-            if "clockId" in self.page:
-                for clock in self.page["clockId"]:
-                    pixoo.set_clock(clock['number'])
-
-            if "images" in self.page:
-                for image in self.page["images"]:
-                    pixoo.draw_image(image['image'], tuple(image['position']))
-
-            if "texts" in self.page:
-                for text in self.page["texts"]:
-                    text_template = Template(text['text'], self.hass)
-                    try:
-                        rendered_text = str(text_template.async_render())
-                    except TemplateError as e:
-                        _LOGGER.error("Template render error: %s", e)
-                        rendered_text = "Template Error"
-
-                    if text['font'] == "FONT_PICO_8":
-                        pixoo.draw_text(rendered_text, tuple(text['position']), tuple(text['font_color']), FONT_PICO_8)
-                    if text['font'] == "FONT_GICKO":
-                        pixoo.draw_text(rendered_text.upper(), tuple(text['position']), tuple(text['font_color']),
-                                        FONT_GICKO)
-                    if text['font'] == "FIVE_PIX":
-                        pixoo.draw_text(rendered_text.upper(), tuple(text['position']), tuple(text['font_color']), FIVE_PIX)
-
-            if "PV" in self.page:
-                solar(pixoo, self.hass, self.page, FONT_PICO_8, FONT_GICKO)
-
-            if "Fuel" in self.page:
-                _LOGGER.info(f"Fuel = {self.page}")
-                _LOGGER.info(f"Fuel = {self.page['Fuel']}")
-                fuel(pixoo, self.hass, self.page, FONT_PICO_8, FONT_GICKO, FIVE_PIX, ELEVEN_PIX)
-
-            if "channel" not in self.page and "clockId" not in self.page:
-                pixoo.push()
+        if len(self._pages) == 0:
+            return
 
-        page_displayed = False
-        initial_index = self._current_page_index
+        is_enabled = None
+        iteration_count = 0
+        self._current_page_index = (self._current_page_index + 1) % len(self._pages)  # Increment the page index, duh.
+        while not is_enabled:
+            if iteration_count >= len(self._pages):
+                _LOGGER.info("All pages disabled. Not updating.")
+                break
 
-        while not page_displayed:
             self.page = self._pages[self._current_page_index]
-            self.schedule_update_ha_state()
 
             try:
-                is_enabled = str(Template(self.page['enable'], self.hass).async_render()) if 'enable' in self.page else "true"
+                is_enabled = str(Template(str(self.page.get('enabled', 'true')), self.hass).async_render())
```

Comment:
I'm not sure... You see, with templates, there's something that's called "is_state" (for example, is_state('device_tracker.paulus', 'home')) which returns true or false. I think it would be best to assume a true or false input instead of an entity's state, since if we expect the latter, we would have to handle all sorts of scenarios, like for example, playing or not for media players, etc... and I don't think that this is a good idea for the scopes of this integration. Therefore, I think we should stick to true/false and their alternatives (on, 1, true, yes, and so on)

Reply:
yes, you're right, it would take on an insane dimension. to be honest, i hadn't thought that far ahead